OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -88,29 +103,44 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
             }
         }
 
+        let notShowingCurrentTickle = self.currentMessage != nil
+            && self.currentMessage!.id != (self.messageQueue[0] as! InAppMessage).id
+            && (self.messageQueue[0] as! InAppMessage).id == self.pendingTickleIds[0] as! Int64
+
+        let queueNotEmptyAndNotShowingAnything = self.currentMessage == nil && self.messageQueue.count > 0
+
+        let shouldShowSomething = notShowingCurrentTickle || queueNotEmptyAndNotShowingAnything
+
         messageQueueLock.signal()
-      
-        DispatchQueue.main.async {
-            self.initViews()
```

Comment:
Confirmed, the JS evaluation on the first time load will result in the following error:

```
Optional(Error Domain=WKErrorDomain Code=4 "A JavaScript exception occurred" UserInfo={WKJavaScriptExceptionLineNumber=1, WKJavaScriptExceptionMessage=ReferenceError: Can't find variable: postHostMessage, WKJavaScriptExceptionColumnNumber=16, WKJavaScriptExceptionSourceURL=undefined, NSLocalizedDescription=A JavaScript exception occurred})
```

It doesn't seem to cause any fault, but in Android there was a refactor to hold a state machine over the view, perhaps we should introduce some 'ready state' tracking here as well?

Reply:
could we init views only if state is not paused? then also quit before showing spinner if it became paused on next present

`unnecessary and undesirable to couple view init & initial presentation together.` -- it's still coupled, but with + failing call before. And in any case it will remain coupled if we need to wait for 'ready'

Reply:
Yes, we already won't init views (now) if presentation is paused, this is a slightly different condition that was exposed by the work.

In Android SDK you can see how we handle this in the InAppView, it maintains a state machine over 'loading' -> 'ready' -> 'disposed', and it will only pass messages to the renderer once ready is achieved.

Here we don't have the distinctions and the refactoring results in us naively trying to pass the message to the view.

We could potentially explore a similar state machine or 'ready' flag for the webview and skip passing messages if not ready, btu I'm not sure if practically this error affects much.

WDYT?

Reply:
Have added a simple ready state flag to track webview initialisation and prevent this error.